docs/content/contributing/Networking.md

```diff
@@ -0,0 +1,60 @@
+# Networking
+
+We using [Okhttp](https://square.github.io/okhttp/) sometimes with [Retrofit](https://square.github.io/retrofit/) for
+networking in gradle plugins and android.
+
+## Obtain an OkHttpClient
+
+Add dependency on `http-client` module:
+
+```kotlin
+dependencies {
+    implementation(project(":common:http-client"))
+}
+```
+
+Create an instance of `HttpClientProvider`
+
+```kotlin
+val httpClientProvider = HttpClientProvider(statsdSender, timeProvider)
+```
+
+Get client
+
+```kotlin
+val httpClient = httpClientProvider.provide().build()
+```
+
+Method provide() returns `OkHttpClient.Builder` so you can configure it further
+
+## Required tag
+
+Obtained client needs some additional information to be able to gather required statistics.
+
+Every request should contain a tag of type `RequestMetadata`
+
+???+ warning 
+    Missing tag will lead to missing metrics for this api method / service 
+    Warning message could be found in logs
```

Comment:
No one reads logs. Can't we make it fail-fast?

Reply:
The alternative is an opt-out if the client doesn't need metrics for a method.

Reply:
I propose to send metric with `no-tag` it will help us to see it on dashboard

Reply:
> Can't we make it fail-fast?

fail fast is prohibited i event listener

https://square.github.io/okhttp/4.x/okhttp/okhttp3/-event-listener/
```
All event methods must execute fast, without external locking, cannot throw exceptions, attempt to mutate the event parameters, or be re-entrant back into the client. Any IO - writing to files or network should be done asynchronously.
```

> I propose to send metric with no-tag it will help us to see it on dashboard

i like it, will do